src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -394,6 +395,28 @@ IHeaderDictionary IHttpResponseTrailersFeature.Trailers
             set => ResponseTrailers = value;
         }
 
+        internal IHttpResetFeature GetResetFeature()
+        {
+            // Check version is above 2.
+            if (HttpVersion >= System.Net.HttpVersion.Version20 && NativeMethods.HttpSupportTrailer(_pInProcessHandler))
+            {
+                return this;
+            }
+
+            return null;
+        }
+
+        void IHttpResetFeature.Reset(int errorCode)
+        {
+            SetResetCode(errorCode);
```

Comment:
Validate that errorCode isn't negative? I don't know if we did that in Kestrel or HttpSys

Reply:
https://http2.github.io/http2-spec/#ErrorCodes

Size can be up to 32 bit so there is no need to check for an upper limit.

Reply:
I don't this HttpSys does, but I can add that check.